### ISSUE TITLE
we may want to prefix https:// in front of OpenID login hostname requests

### ISSUE THREAD

**Issue by kareila** [dreamwidth/dw-free#2807]
I tried logging into wiki.dreamwidth.net by typing in `kareila.dreamwidth.org` as my OpenID username, and it failed. Going back and typing in `https://kareila.dreamwidth.org` instead worked as expected.

**Comment by kareila** [comment]
@alierak says that specifying anything other than `http://` in front of a bare hostname is a violation of the OpenID spec and that he would prefer to change what the wiki sends to DW than to break our implementation of the spec (again).

**Comment by alierak** [comment]
No, I said that an unqualified hostname refers specifically to the http protocol and can't be upgraded while still matching the requested identity.

**Comment by alierak** [comment]
When I try to log into the wiki as alierak.dreamwidth.org, the error I get is "The site you just came from wants to verify an identity that you can't provide as [personal profile] alierak." If I take the URL of that error page, and remove references to port 443 from a couple of its parameters, the authentication attempt succeeds. Those parameters are:

identity=https%3A%2F%2Falierak.dreamwidth.org%3A443%2F

Obviously, just remove the %3A443.

And:

return_to=http%3A%2F%2Fwiki.dreamwidth.net%2Fwiki ... more stuff here ... zi8yqT%26openid1_claimed_id%3Dhttps%253A%252F%252Falierak.dreamwidth.org%253A443%252F

So there, I've got to remove %253A443 near the end.

If you remove the 443 from the identity parameter, DW's part of the authentication succeeds, but after you return to the wiki it will notice that the 443 was removed and the identities don't match. So you also have to remove it from the openid1_claimed_id portion of the return_to URL in order to the wiki to notice that the authentication attempt and result match.

Where is it getting the :443 in the first place? Well, it first tries to look up the openid endpoint over plain HTTP like I told it to by not specifying a protocol. When it gets redirected, that's using the redirect rule I added to the prod ELB instance. It wouldn't let me save the rule if I didn't specify a port, even though "redirect to HTTPS" has an obvious commonly accepted meaning that doesn't involve adding ":443" to the output. But ELB does it, so the wiki gets back a location for my openid endpoint that contains ":443", and sends me over to DW to authenticate with the stuff shown above.

It would be nice if we could generate the redirect to HTTPS without the :443.

I guess we could address this in the openid controller by specifically editing identity and return_to as shown above, for dreamwidth.org:443 identities. I don't know if openid1_claimed_id (part of the wiki's return_to URL) is any kind of standard thing that you should expect to be able to find and edit for openid requests coming from sites other than the wiki.

Another option might be to see if we can get DW to successfully authenticate a request with the :443 in it, rather than calling it "an identity that you can't provide".
